File: fetch_service/src/main/java/ru/tagmeasurements/fetch_service/models/TemperatureZone.java
package ru.tagmeasurements.fetch_service.models;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.*;
import java.util.List;

@Data
@Entity
@NoArgsConstructor
@AllArgsConstructor
public class TemperatureZone {
    @Id
    private Long id;
    private String name;
    private String description;
    private Double lowerTempLimit;
    private Double higherTempLimit;
    private String notifyEmails;
    @ManyToMany
    @JoinTable(
            name = "temperature_zone_tags",
            joinColumns = @JoinColumn(name = "temperature_zone_id"),
            inverseJoinColumns = @JoinColumn(name = "tag_uuid")
    )
    private List<Tag> tags;
}

File: pkg/models/TemperatureZone.go
package models

type TemperatureZone struct {
	ID              uint    `json:"id" gorm:"type:serial;primary_key"`
	Name            string  `json:"name" gorm:"unique;not_null"`
	Description     string  `json:"description"`
	LowerTempLimit  float64 `json:"lower_temp_limit"`
	HigherTempLimit float64 `json:"higher_temp_limit"`
	NotifyEmails    string  `json:"notify_emails"`
	Tags            []Tag   `json:"tags" gorm:"many2many:temperature_zone_tags;"`
}

func (TemperatureZone) TableName() string {
	return "temperature_zone"
}
